k8s/provider.go

```diff
@@ -171,3 +158,66 @@ func (p *ServiceProvider) getPodWithRetry(ctx context.Context, client Client, po
 		return nil, fmt.Errorf("could not get valid Pod data after %s", p.Timeout)
 	}
 }
+
+func generateServices(serviceName string, pod *corev1.Pod, globalTags []string) ([]consul.ServiceInstance, error) {
+	portDefinitions, err := getPortDefinitions()
+	if err != nil {
+		return nil, err
+	}
+
+	if portDefinitions == nil {
+		return generateFromContainerPorts(serviceName, pod, globalTags)
+	}
+	return generateFromPortDefinitions(serviceName, portDefinitions, pod, globalTags)
+}
+
+func generateFromContainerPorts(serviceName string, pod *corev1.Pod, globalTags []string) ([]consul.ServiceInstance, error) {
+	// this method exists only to provide backward compatibility
+	// TODO(medzin): Allow to specify which containers and ports will be registered
```

Comment:
This TODO can be removed I think, as the `generateFromPortDefinitions` method does just this.